compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_instruction_gen.bal

```diff
@@ -335,6 +339,59 @@ type InstructionGenerator object {
         self.mv.visitInsn(POP);
     }
 
+    # Generate a new instance of an array value
+    # 
+    # + inst - type of the new array
+    function generateArrayNewIns(bir:NewArray inst) {
+        self.mv.visitTypeInsn(NEW, ARRAY_VALUE);
+        self.mv.visitInsn(DUP);
+        
+        bir:BType arrayType = inst.typeValue;
+        if (arrayType is bir:BArrayType) {
+            loadType(self.mv, arrayType.eType);
+            self.mv.visitVarInsn(LLOAD, self.getJVMIndexOfVarRef(inst.sizeOp.variableDcl));
+            self.mv.visitMethodInsn(INVOKESPECIAL, ARRAY_VALUE, "<init>", io:sprintf("(L%s;J)V", BTYPE), false);
+        }
+        self.mv.visitVarInsn(ASTORE, self.getJVMIndexOfVarRef(inst.lhsOp.variableDcl));
+    }
+    # Generate adding a new value to array
+    function generateArrayStoreIns(bir:ArrayStore inst) {
+        // TODO: visit(var_ref)
+        // TODO: visit(index_expr)
+        // TODO: visit(value_expr)
+        int varRefIndex = self.getJVMIndexOfVarRef(inst.lhsOp.variableDcl);
+        self.mv.visitVarInsn(ALOAD, varRefIndex);
+        int keyIndex = self.getJVMIndexOfVarRef(inst.keyOp.variableDcl);
+        self.mv.visitVarInsn(LLOAD, keyIndex);
+        // int valueIndex = self.getJVMIndexOfVarRef(inst.rhsOp.variableDcl);
+        // self.mv.visitVarInsn(LLOAD, valueIndex); //need to fix, this is only for int
+        int valueIndex = self.getJVMIndexOfVarRef(inst.rhsOp.variableDcl);
+        bir:BType valueType = inst.rhsOp.variableDcl.typeValue;
+        self.loadFromLocalVar(valueType, valueIndex);
+
+        string valueDesc = getMethodArgDesc(valueType); //pass the value type
+        self.mv.visitMethodInsn(INVOKEVIRTUAL, ARRAY_VALUE, "add", io:sprintf("(J%s)V", valueDesc), false);
+    }
+
+    function generateArrayValueLoad() {
```

Comment:
Are we using this function in any place?

Reply:
i think it will be used when we have array load instruction

Reply:
Yes, not used ATM. Need to revisit once the array load available from BIR.